opennlp-tools/src/main/java/opennlp/tools/util/TrainingParameters.java

```diff
@@ -37,26 +37,50 @@
   public static final String CUTOFF_PARAM = "Cutoff";
   public static final String THREADS_PARAM = "Threads";
 
-  private Map<String, String> parameters = new HashMap<>();
+  private Map<String, Object> parameters = new HashMap<>();
 
   public TrainingParameters() {
   }
 
   public TrainingParameters(TrainingParameters trainingParameters) {
     this.parameters.putAll(trainingParameters.parameters);
   }
-  
+
   public TrainingParameters(Map<String,String> map) {
-    parameters.putAll(map);
+    //parameters.putAll(map);
+    // try to respect their original type...
+    for (String key: map.keySet()) {
```

Comment:
Thanks for the comment. I think it's an interesting idea, but I'd like to stick with my code because this is just transient stuff for back-compat reason and in the future version, we will throw this constructor away which takes Map<String,String> and have another one which takes Map<String,Object> instead.

So I think I have to add deprecated flag to it and the new constructor which is commented out in the next PR.